src/features/earn/providers/yearn/api/api.ts

```diff
@@ -107,31 +109,53 @@ export class YearnVaultApi {
     return await this.web3.eth.getTransactionReceipt(txid)
   }
 
+  checksumAddress(address: string): string {
+    return this.web3.utils.toChecksumAddress(address)
+  }
+
+  async getVaultId({
+    tokenContractAddress,
+    vaultAddress
+  }: {
+    tokenContractAddress: string
+    vaultAddress: string
+  }): Promise<number> {
+    const routerAddress = new this.web3.eth.Contract(ssRouterAbi, ssRouterContractAddress)
```

Comment:
nit: would rename variable here to something more clear.

Reply:
Yeah, that was misnamed, it's the router contract, not the address. I'm brought the contract into the constructor and am using that now.